crates/cxx-qt-gen/src/generator/rust/signals.rs

```diff
@@ -11,11 +11,137 @@ use crate::{
+            .get(&signal.qobject_ident.to_string()),
+    );
+
+    let parameters_cxx: Vec<FnArg> = signal
+        .parameters
+        .iter()
+        .map(|parameter| {
+            let ident = &parameter.ident;
+            let ty = &parameter.ty;
+            parse_quote! { #ident: #ty }
+        })
+        .collect();
+    let parameters_qualified: Vec<FnArg> = parameters_cxx
+        .iter()
+        .cloned()
+        .map(|mut parameter| {
+            if let FnArg::Typed(pat_type) = &mut parameter {
+                *pat_type.ty =
+                    syn_type_cxx_bridge_to_qualified(&pat_type.ty, &cxx_mappings.qualified);
+            }
+            parameter
+        })
+        .collect();
+
+    let self_type_cxx = if signal.mutable {
+        parse_quote! { Pin<&mut #qobject_name> }
+    } else {
+        parse_quote! { &#qobject_name }
+    };
+    let self_type_qualified =
+        syn_type_cxx_bridge_to_qualified(&self_type_cxx, &cxx_mappings.qualified);
+    let qualified_impl =
+        syn_ident_cxx_bridge_to_qualified_impl(qobject_name, &cxx_mappings.qualified);
+
+    let mut unsafe_block = None;
+    let mut unsafe_call = Some(quote! { unsafe });
+    if signal.safe {
+        std::mem::swap(&mut unsafe_call, &mut unsafe_block);
+    }
+
+    let attrs = &signal.method.attrs;
+
+    let fragment = RustFragmentPair {
+        cxx_bridge: vec![
+            quote! {
+                #unsafe_block extern "C++" {
+                    #(#attrs)*
+                    #[rust_name = #signal_name_rust_str]
+                    #unsafe_call fn #signal_name_cpp(self: #self_type_cxx, #(#parameters_cxx),*);
+                }
+            },
+            quote! {
+                unsafe extern "C++" {
+                    #[doc(hidden)]
+                    #[namespace = #connect_namespace]
+                    #[must_use]
+                    #[rust_name = #free_connect_ident_rust_str]
+                    fn #free_connect_ident_cpp(self_value: #self_type_cxx, func: #unsafe_call fn(#self_type_cxx, #(#parameters_cxx),*), conn_type: CxxQtConnectionType) -> CxxQtQMetaObjectConnection;
+                }
+            },
+        ],
+        implementation: vec![
+            quote! {
+                impl #qualified_impl {
+                    #[doc = "Connect the given function pointer to the signal "]
+                    #[doc = #signal_name_cpp_str]
+                    #[doc = ", so that when the signal is emitted the function pointer is executed."]
+                    #[doc = "\n"]
+                    #[doc = "Note that this method uses a AutoConnection connection type."]
+                    #[must_use]
+                    pub fn #on_ident_rust(self: #self_type_qualified, func: fn(#self_type_qualified, #(#parameters_qualified),*)) -> cxx_qt_lib::QMetaObjectConnection
+                    {
+                        #module_ident::#free_connect_ident_rust(self, func, cxx_qt_lib::ConnectionType::AutoConnection)
```

Comment:
Note for future:
ConnectionType needs to be in cxx_qt, not cxx_qt_lib.

Reply:
Yup this would be as part of #319 and once we have packed_bundled_libs support in rustc i think this becomes easier as well.